test/react-web/client/graphql/queries.test.tsx

```diff
@@ -484,14 +484,42 @@ describe('queries', () => {
     @graphql(query, {
       skip: ({ skip }) => skip,
       options: ({ willThrowIfAccesed }) => ({ pollInterval: willThrowIfAccesed.pollInterval }),
-      props: ({ willThrowIfAccesed }) => ({ pollInterval: willThrowIfAccesed.pollInterval }),
     })
     class Container extends React.Component<any, any> {
       componentWillReceiveProps(props) {
         queryExecuted = true;
       }
       render() {
-        expect(this.props.data).toBeFalsy();
+        expect(this.props.data).toEqual({skipped: true});
+        return null;
+      }
+    };
+
+    renderer.create(<ApolloProvider client={client}><Container skip={true} /></ApolloProvider>);
+
+    setTimeout(() => {
+      if (!queryExecuted) { done(); return; }
+      fail(new Error('query ran even though skip present'));
+    }, 25);
+  });
+
+  it('runs props even when skipped', (done) => {
+    const query = gql`query people { allPeople(first: 1) { people { name } } }`;
+    const data = { allPeople: { people: [ { name: 'Luke Skywalker' } ] } };
+    const networkInterface = mockNetworkInterface({ request: { query }, result: { data } });
+    const client = new ApolloClient({ networkInterface, addTypename: false });
+
+    let queryExecuted;
+    @graphql(query, {
+      skip: ({ skip }) => skip,
+      props: ({ data }) => ({ skippery: data.skipped })
```

Comment:
Could we add this assertion here to make sure `data` is correct in the `props` function?

```js
expect(this.props.data).toEqual({skipped: true});
```